src/Api/ApiResourceRegistrar.php

```diff
@@ -45,96 +66,244 @@ public function __construct(Router $api)
-            array_splice($arguments, 0, 0, [$name]);
-            return call_user_func_array([$this, 'action'], $arguments);
-        }
-        throw new ConfigurationException("Unknown HTTP verb $name used for route $arguments[0]");
+        $reflectionClass = new \ReflectionClass($modelClass);
+        return $reflectionClass->getShortName();
     }
 
     /**
-     * Actually called method, when user calls verb methods
+     * Actually called method, when user calls verb methods.
      *
      * @param string $verb - one of GET / POST / PUT / DELETE
-     * @param string $path - URL path
+     * @param string $path URL path
      * @param string $controller Class, containing action method
-     * @param string|null $action method, which will be executed on route hit
-     * @param string|null $route - route name
+     * @param string|null $action Method, which will be executed on route hit
+     * @param string|null $routeName Route name
+     * @param array $mapping Model bindings mapping
+     *
      * @return mixed
      */
-    private function action(string $verb, string $path, string $controller, string $action = null, string $route = null)
-    {
+    private function action(
+        string $verb,
+        string $path,
+        string $controller,
+        ?string $action = null,
+        ?string $routeName = null,
+        array $mapping = []
+    ) {
         $pos = strrpos($path, '/', -1);
-        $pathLastSegment = $pos ? substr($path, $pos+1) : $path;
-
+        $pathLastSegment = $pos ? substr($path, $pos + 1) : $path;
         if (!$action) {
             $action = $pathLastSegment;
         }
-        if (!$route) {
-            $route = strtolower(str_replace('/', '.', $path));
+        if (!$routeName) {
+            $routeName = strtolower(str_replace('/', '.', $path));
             // Small piece of magic: make auto-named routes look nicer
-            if ($pathLastSegment != $action) {
-                if (strrpos($route, '.'.$pathLastSegment, -1) === false) {
-                    $route = "$route.$action";
+            if ($pathLastSegment !== $action) {
+                if (strrpos($routeName, '.' . $pathLastSegment, -1) === false) {
+                    $routeName = "$routeName.$action";
                 } else {
-                    $route = str_replace('.' . $pathLastSegment, '.' . $action, $route);
+                    $routeName = str_replace('.' . $pathLastSegment, '.' . $action, $routeName);
                 }
             }
-            $route = strtolower($route);
+            $routeName = strtolower($routeName);
         }
-        return $this->api->$verb($path ?: $action, ['uses' => $controller.'@'.$action, 'as' => $route]);
+        return $this->api->$verb(
+            $path ?? $action,
+            ['uses' => "$controller@$action", 'as' => $routeName, 'mapping' => $mapping]
+        );
     }
 
-    private function asArray($value): array
+    /**
+     * Converts params to needed form.
```

Comment:
Not clear, what is **needed form** without analyzing what was passed and what was made